packages/hardhat-test-helpers/src/utils.ts

```diff
@@ -67,3 +67,22 @@ export function toRpcQuantity(x: NumberLike): string {
 
   return hex.startsWith("0x") ? hex.replace("0x0", "0x") : `0x${hex}`;
 }
+
+export function assertValidAddress(hexAddress: string): void {
```

Comment:
This function also has to verify the checksum of the address, unless the address is all lowercase.

Reply:
Are you sure? I just tested it and it doesn't seem like our RPC methods nor the official EVM methods reject a non-checksum (or lowercase) address parameter. I just tested with an all capitalized address as an example.

Reply:
Yes, but the test helpers don't have to have the same level of strictness that the RPC layer. In some cases we are being stricter (e.g., throwing when an invalid snapshot is used) and in others we are being more relaxed (accepting more kinds of number values). In this case I think it makes sense to add the extra check, since it's a common expectation.